core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -71,11 +73,16 @@ class TaskInfo(
     failed = true
   }
 
+  private[spark] def markKilled(time: Long = System.currentTimeMillis) {
```

Comment:
Can you consolidate this and the two methods above into a single markFinished method that accepts a TaskState and a time? And then that method can handle matching the TaskState to the appropriate value of killed / failed.